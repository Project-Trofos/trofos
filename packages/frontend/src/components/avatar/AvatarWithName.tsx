import React from 'react';
import { Avatar as AntdAvatar } from 'antd';

import './AvatarWithName.css';

export type AvatarProps = {
  username: string | React.ReactNode;
  icon: React.ReactNode;
};

export default function AvatarWithName(props: AvatarProps) {
  const { icon, username } = props;
  return (
    <div className="avatar-container">
      <AntdAvatar icon={icon} />
      <div>{username}</div>
    </div>
  );
}
