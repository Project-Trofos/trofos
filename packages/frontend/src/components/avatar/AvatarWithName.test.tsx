import React from 'react';
import { render, screen } from '@testing-library/react';
import AvatarWithName from './AvatarWithName';

describe('test Avatar', () => {
  describe('when rendering', () => {
    it('should render correctly', () => {
      const ICON = <div>icon</div>;
      const USERNAME = 'username1';
      const { baseElement } = render(<AvatarWithName icon={ICON} username={USERNAME} />);

      // Ensure fields are present
      expect(screen.getByText('icon')).toBeInTheDocument();
      expect(screen.getByText(USERNAME)).toBeInTheDocument();

      // Compare with snapshot to ensure structure remains the same
      expect(baseElement).toMatchSnapshot();
    });
  });
});
