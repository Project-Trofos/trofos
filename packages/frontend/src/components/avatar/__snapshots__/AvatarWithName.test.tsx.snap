// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test Avatar > when rendering > should render correctly 1`] = `
<body>
  <div>
    <div
      class="avatar-container"
    >
      <span
        class="ant-avatar ant-avatar-circle ant-avatar-icon"
      >
        <div>
          icon
        </div>
      </span>
      <div>
        username1
      </div>
    </div>
  </div>
</body>
`;
